src/app/components/List/EBookList.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import appConfig from '../../../../appConfig';
+
+const ebookUrl = appConfig.ereader[process.env.APP_ENV];
+
+/**
+ * Create a link defaulting to a ebook "download" unless
+ * the ebook is hosted by us (link has our S3 bucket pattern)
+ * which shows a link to our eReader with the S3 bucket URL as
+ * a query parameter.
+ * @param {string} url
+ * @return {string}
+ */
+const generateLink = (url) => {
+  let link = `${url}`;
+  let linkText = 'Download';
+
+  if (url.includes('simplye')) {
+    link = `${ebookUrl}?url=${url}`;
+    linkText = 'Read Online';
+  }
+
+  return (
+    <a href={`${link}`}>{`${linkText}`}</a>
+  );
+};
+
+const EBookList = (ebooks) => {
+  return (
+    <ul className="nypl-items-list">
+      {ebooks.ebooks.map((ebook, ebookKey) => (
+        <li key={ebookKey.toString()}>
+          {generateLink(ebook.url)} [{ebook.epub_path.split('/').pop()}]
```

Comment:
Is the `epub_path` display temporary? That info doesn't seem readily useful to me.

Reply:
This is temporary to distinguish one book from another by filename since we're giving the same link name "Read Online" or "Download" which isn't descriptive. Gutenberg tends to have a version with images and one without. Needs a different treatment.